src/resource/sid.c

```diff
@@ -45,7 +48,10 @@ static int _init_sid(sid_resource_t *res, const void *kickstart_data, void **dat
 	}
 
 	if (sid_resource_create_signal_event_source(res, NULL, SIGTERM, NULL, 0, "sigterm", NULL) < 0 ||
-	    sid_resource_create_signal_event_source(res, NULL, SIGINT, NULL, 0, "sigint", NULL) < 0) {
+	    sid_resource_create_signal_event_source(res, NULL, SIGINT, NULL, 0, "sigint", NULL) < 0 ||
+	    sid_resource_create_signal_event_source(res, NULL, SIGPIPE, NULL, 0, "sigpipe", NULL) < 0 ||
+	    sid_resource_create_signal_event_source(res, NULL, SIGHUP, NULL, 0, "sighup", NULL) < 0 ||
+	    sid_resource_create_signal_event_source(res, NULL, SIGCHLD, NULL, 0, "sigchld", NULL) < 0) {
```

Comment:
...so here we should to use a handler for `SIGTERM` with the `sid_resource_exit_event_loop` call.

Reply:
yes, ok - that would be better

Reply:
The function signature for sid_resource_exit_event_loop():

int sid_resource_exit_event_loop(sid_resource_t *res)

The signal handler is:

int _sid_signal_handler(sid_resource_event_source_t *es, const struct signalfd_siginfo *si, void *data)

I updated sid_resource_exit_event_loop() to take sid_resource_event_source_t * as the first parameter.  I'm not sure that was what you were looking to do.  Would it be better to do the parameter translation in the new _sid_signal_handler()?

Reply:
Or we can simply pass the resource as the custom arg to the signal handler function like here https://github.com/sid-project/sid/blob/f64443e69f3a824ea1433debcf9ad98bebaeed3b/src/resource/worker-control.c#L1098

For the `sid` resource (`src/resource/sid.c`), that could look something like this:

```
static int _on_sid_signal_event(sid_resource_event_source_t *es, const struct signalfd_siginfo *si, void *arg)
{
  sid_resource_t *res = arg;
  sid_resoruce_exit_event_loop(res);
}

....

static int _init_sid(sid_resource_t *res, const void *kickstart_data, void **data)
{
...
sid_resource_create_signal_event_source(res, NULL, SIGTERM, NULL, 0, "sigterm", res);
...
}
```

I've already used it this way in

Reply:
Thanks.  That is much better.